### PR TITLE
RCAL-193: Update Mask Code in Ramp, Flat, and Jump step and tests to RDM

### PR DESCRIPTION
**Pull request by PaulHuwe**
Updated rampfit and flat maker utilities to support the same functionality as the other model maker functions. 
Streamlined and commented all maker utility functions. 
Added tests to complete coverage of roman_datamodels/testing/utils.py
Cleaned out some deprecated code.

### CODE REVIEW

File: src/roman_datamodels/testing/utils.py [hunk 1]
```diff
@@ -340,20 +431,135 @@ def add_ref_common(meta):
...
+    """
     meta = {}
     add_ref_common(meta)
     flatref = stnode.FlatRef()
     meta['reftype'] = 'FLAT'
     flatref['meta'] = meta
+
+    if not shape:
+        shape = (4096, 4224)
+
```
Comment (ddavis-stsci) [inline src/roman_datamodels/testing/utils.py:534]: The current flat are 4096x 4096 and we should update the default to match.

File: src/roman_datamodels/testing/utils.py [hunk 2]
```diff
@@ -366,14 +572,12 @@ def mk_dark(shape=None, filepath=None):
...
-        darkref['data'] = np.zeros(shape, dtype=np.float32)
-        darkref['dq'] = np.zeros(shape[1:], dtype=np.uint32)
-        darkref['err'] = np.zeros(shape, dtype=np.float32)
-    else:
-        darkref['data'] = np.zeros((7, 4096, 4224), dtype=np.float32)
-        darkref['dq'] = np.zeros((4096, 4224), dtype=np.uint32)
-        darkref['err'] = np.zeros((7, 4096, 4224), dtype=np.float32)
+    if not shape:
+        shape = (2, 4096, 4224)
+
```
Comment (ddavis-stsci) [inline src/roman_datamodels/testing/utils.py:577]: update 4096x4096 for the default.

File: src/roman_datamodels/testing/utils.py [hunk 3]
```diff
@@ -384,16 +588,32 @@ def mk_dark(shape=None, filepath=None):
...
     meta['reftype'] = 'GAIN'
     gainref['meta'] = meta
 
-    if shape:
-        gainref['data'] = np.zeros(shape, dtype=np.float32)
-    else:
-        gainref['data'] = np.zeros((4096, 4224), dtype=np.float32)
+    if not shape:
+        shape = (4096, 4224)
+
```
Comment (ddavis-stsci) [inline src/roman_datamodels/testing/utils.py:615]: 4224 --> 4096

File: src/roman_datamodels/testing/utils.py [hunk 4]
```diff
@@ -403,18 +623,33 @@ def mk_gain(shape=None, filepath=None):
...
 
-    if shape:
-        linearityref['coeffs'] = np.zeros(shape, dtype=np.float32)
-        linearityref['dq'] = np.zeros(shape[1:], dtype=np.uint32)
-    else:
-        linearityref['coeffs'] = np.zeros((2, 4096, 4224), dtype=np.float32)
-        linearityref['dq'] = np.zeros((4096, 4096), dtype=np.uint32)
+    if not shape:
+        shape = (2, 4096, 4224)
+
```
Comment (ddavis-stsci) [inline src/roman_datamodels/testing/utils.py:650]: 4224 --> 4096

File: src/roman_datamodels/testing/utils.py [hunk 5]
```diff
@@ -502,10 +798,10 @@ def mk_readnoise(shape=None, filepath=None):
     exposure['type'] = 'WFI_IMAGE'
     readnoiseref['meta']['exposure'] = exposure
 
-    if shape:
-        readnoiseref['data'] = np.zeros(shape, dtype=np.float32)
-    else:
-        readnoiseref['data'] = np.zeros((4096, 4224), dtype=np.float32)
+    if not shape:
+        shape = (4096, 4224)
+
```
Comment (ddavis-stsci) [inline src/roman_datamodels/testing/utils.py:803]: 4224 to 4096

File: src/roman_datamodels/testing/utils.py [hunk 6]
```diff
@@ -564,18 +890,33 @@ def mk_rampfitoutput(arrays=True, filepath=None):
...
 
-    if shape:
-        saturationref['data'] = np.zeros(shape, dtype=np.float32)
-        saturationref['dq'] = np.zeros(shape, dtype=np.uint32)
-    else:
-        saturationref['data'] = np.zeros((4096, 4224), dtype=np.float32)
-        saturationref['dq'] = np.zeros((4096, 4224), dtype=np.uint32)
+    if not shape:
+        shape = (4096, 4224)
+
```
Comment (ddavis-stsci) [inline src/roman_datamodels/testing/utils.py:917]: 4224 --> 4096

File: src/roman_datamodels/testing/utils.py [hunk 7]
```diff
@@ -585,20 +926,34 @@ def mk_saturation(shape=None, filepath=None):
...
-        superbiasref['data'] = np.zeros(shape, dtype=np.float32)
-        superbiasref['dq'] = np.zeros(shape, dtype=np.uint32)
-        superbiasref['err'] = np.zeros(shape, dtype=np.float32)
-    else:
-        superbiasref['data'] = np.zeros((4096, 4224), dtype=np.float32)
-        superbiasref['dq'] = np.zeros((4096, 4224), dtype=np.uint32)
-        superbiasref['err'] = np.zeros((4096, 4224), dtype=np.float32)
+    if not shape:
+        shape = (4096, 4224)
+
```
Comment (ddavis-stsci) [inline src/roman_datamodels/testing/utils.py:953]: Update default shape.

**Review by ddavis-stsci — CHANGES_REQUESTED** [review]
It would be good to have all the default shapes consistent.

File: src/roman_datamodels/testing/utils.py [hunk 8]
```diff
@@ -70,6 +93,14 @@ def mk_program():
 
 
 def mk_observation():
+    """
+    Create a dummy Observation instance with valid values for attributes
+    required by the schema. Utilized by the model maker utilities below.
+
+    Returns
+    -------
+    roman_datamodels.stnode.Observation
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply src/roman_datamodels/testing/utils.py:102]: Removed.

File: src/roman_datamodels/testing/utils.py [hunk 9]
```diff
@@ -138,12 +193,28 @@ def mk_photometry():
...
 
 def mk_aperture():
+    """
+    Create a dummy Aperture instance with valid values for attributes
+    required by the schema. Utilized by the model maker utilities below.
+
+    Returns
+    -------
+    roman_datamodels.stnode.Aperture
+    """
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply src/roman_datamodels/testing/utils.py:217]: Removed.

File: src/roman_datamodels/testing/utils.py (hunk 1 above)
Comment (PaulHuwe) [reply src/roman_datamodels/testing/utils.py:534]: All are changed to that shape.

**Review by ddavis-stsci — APPROVED** [review]
Looks good. Let's hope they don't want it back to 4224 next week!

**Pull request merged**